synapse/replication/tcp/streams/_base.py

```diff
@@ -291,9 +291,7 @@ def __init__(self, hs):
         if hs.config.worker_app is None:
             # on the master, query the presence handler
             presence_handler = hs.get_presence_handler()
-            update_function = db_query_to_update_function(
-                presence_handler.get_all_presence_updates
-            )
+            update_function = presence_handler.get_all_presence_updates
```

Comment:
maybe we should get rid of `presence_handler.get_all_presence_updates`, since it's now just a passthrough, and go straight to the storage layer, as many of the other streams do?

Reply:
Yeah, maybe. Though we really we should probably fix the `TODO` and correctly sync in memory updates as well, which means its sort of correct that we're going via the presence handler

Reply:
fair